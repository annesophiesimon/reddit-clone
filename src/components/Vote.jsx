import { BiLike, BiDislike } from 'react-icons/bi';

const Vote = (votes) => {
  const { ups } = votes;
  return (
    <div className="flex text-slate-500">
      <button className="flex items-center mx-2 hover:text-green-700" aria-label="Vote up">
        <BiLike className="mr-2" /> {ups} |
      </button>
      <button aria-label="vote down" className="down hover:text-red-700">
        <BiDislike />
      </button>
    </div>
  );
};

export default Vote;
