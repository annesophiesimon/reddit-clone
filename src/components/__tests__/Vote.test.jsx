import Vote from '../Vote';
import renderer from 'react-test-renderer';

describe('Vote', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Vote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
