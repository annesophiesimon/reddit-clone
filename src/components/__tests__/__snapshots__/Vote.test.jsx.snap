// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vote renders correctly 1`] = `
<div
  className="flex-col items-center text-center mr-2 w-14"
>
  <button
    aria-label="Vote up"
  >
    <svg
      className="hover:fill-green-600"
      fill="currentColor"
      height="24px"
      id="Layer_1"
      version="1.1"
      viewBox="0 0 511.947 511.947"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <g>
          <path
            d="M476.847,216.373L263.513,3.04c-4.267-4.053-10.88-4.053-15.04,0L35.14,216.373c-4.16,4.16-4.16,10.88-0.107,15.04 c2.027,2.027,4.8,3.2,7.573,3.2h128V501.28c0,5.867,4.8,10.667,10.667,10.667h149.333c5.867,0,10.667-4.8,10.667-10.667V234.613 h128c5.867,0,10.667-4.8,10.667-10.667C479.94,221.067,478.873,218.4,476.847,216.373z M330.607,213.28 c-5.867,0-10.667,4.8-10.667,10.667v266.667h-128V223.947c0-5.867-4.8-10.667-10.667-10.667H68.42L255.94,25.547L443.567,213.28 H330.607z"
          />
        </g>
      </g>
    </svg>
  </button>
  <p />
  <button
    aria-label="vote down"
    className="down"
  >
    <svg
      className="hover:fill-red-600"
      fill="currentColor"
      height="24px"
      id="Layer_1"
      version="1.1"
      viewBox="0 0 512.027 512.027"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <g>
          <path
            d="M479.114,283.84c-1.707-3.947-5.547-6.507-9.813-6.507h-128V10.667C341.3,4.8,336.5,0,330.633,0H181.3 c-5.867,0-10.667,4.8-10.667,10.667v266.667h-128c-5.867,0-10.667,4.8-10.56,10.773c0,2.773,1.067,5.44,3.093,7.36L248.5,508.907 c4.16,4.16,10.88,4.16,15.04,0l213.333-213.44C479.86,292.373,480.82,287.893,479.114,283.84z M255.967,486.293L68.34,298.667 H181.3c5.867,0,10.667-4.8,10.667-10.667V21.333h128V288c0,5.867,4.8,10.667,10.667,10.667h112.96L255.967,486.293z"
          />
        </g>
      </g>
    </svg>
  </button>
</div>
`;
